tests/test_matrix_utils_connect.py

```
@@ -107,7 +115,32 @@ async def test_connect_matrix_legacy_config(
 
     # Mock the global matrix_client to None to ensure fresh creation
-    with patch("mmrelay.matrix_utils.matrix_client", None):
+    with (
+        patch("mmrelay.matrix_utils.matrix_client", None),
+        patch(
+            "mmrelay.matrix_utils._resolve_and_load_credentials",
+            Mock(
+                return_value=_ImmediateAwaitable(
+                    MatrixAuthInfo(
+                        homeserver="https://matrix.example.org",
+                        access_token="legacy_token",
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to argument: "access_token"

(S106)